content/blog/teach-tidyverse-2021/index.Rmd

```diff
@@ -0,0 +1,437 @@
+
+slug: teach-tidyverse-2021
+title: Teaching the tidyverse in 2021
+date: 2021-08-31
+author: Mine Çetinkaya-Rundel
+description: >
+    Recommendations for teaching the tidyverse in 2021, summarizing package updates most relevant for teaching data science with the tidyverse, particularly to new learners.
+
+photo:
+  url: https://unsplash.com/photos/lj5ALRcon4g
+  author: Jackie Hope
+
+# one of: "deep-dive", "learn", "package", "programming", or "other"
+categories: [learn] 
+tags: [tidyverse, teaching]
+---
+
+<!--
+TODO:
+* [ x ] Look over / edit the post's title in the yaml
+* [ x ] Edit (or delete) the description; note this appears in the Twitter card
+* [ x ] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [ x ] Find photo & update yaml metadata
+* [ ] Create `thumbnail-sq.jpg`; height and width should be equal
+* [ ] Create `thumbnail-wd.jpg`; width should be >5x height
+* [ ] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+Last summer I wrote a series of blog posts titled [teaching the tidyverse in 2020](https://education.rstudio.com/blog/2020/07/teaching-the-tidyverse-in-2020-part-4-when-to-purrr/).
+As we quickly approach the end of the summer (in the northern hemisphere) and the start of a new academic year, it seems like a good time to provide a new update for teaching the tidyverse, in 2021.
+The main audience for this post is educators who teach the tidyverse and who might want to bring their teaching materials up to date with updates to the tidyverse that happened over the past year.
+Much of what is discussed here has already been covered in package update posts on this blog, but my goal is to summarize the highlights that are most relevant to teaching data science with the tidyverse, particularly to new learners.
+
+Let's get started!
+
+```{r}
+library(tidyverse)
+```
+
+## Lifecycle stages
+
+The [**lifecycle**](https://lifecycle.r-lib.org/) package is used to manage the lifecycle of functions and features within the tidyverse, with clear messaging about what is still experimental and what the tidyverse team is moving away from in the future.
+
+The lifecycle stages you might encounter on functions or packages are experimental, stable, deprecated, and superseded.
+It's helpful to be aware of the stages (and their associated badges) as you review and revise your teaching materials or as you consider incorporating new tooling into your teaching.
+
+![A diagram showing the transitions between the four main stages: experimental can become stable and stable can become deprecated or superseded.](lifecycle.png)
+
+-   <img src="lifecycle-stable.svg" alt="Stable" style="vertical-align:middle"/> Stable indicates that breaking changes will be avoided where possible, and they're only made if the long term benefit of such a change exceeds the short term pain of changing existing code.
```

Comment:
I think you could emphasise a little more strongly that this is the default state for most functions in the tidyverse.